SpatialGDK/Source/SpatialGDK/Public/SpatialGDKSettings.h

```diff
@@ -70,29 +70,55 @@ class SPATIALGDK_API UEventTracingSamplingSettings : public UObject
 {
 	GENERATED_BODY()
 public:
+	struct TraceQueryDeleter
+	{
+		void operator()(Trace_Query* Query) const
+		{
+			if (Query != nullptr)
+			{
+				Trace_Query_Destroy(Query);
+			}
+		}
+	};
+
+	using TraceQueryPtr = TUniquePtr<Trace_Query, TraceQueryDeleter>;
+
 	UPROPERTY(EditAnywhere, Category = "Event Tracing", meta = (ClampMin = 0.0f, ClampMax = 1.0f))
 	double SamplingProbability = 1.0f;
 
 	UPROPERTY(EditAnywhere, Category = "Event Tracing")
 	TMap<FName, double> EventSamplingModeOverrides;
 
 	UPROPERTY(EditAnywhere, Category = "Event Tracing")
-	FString GDKEventPreFilter;
+	FString GDKEventPreFilter = "false";
 
 	UPROPERTY(EditAnywhere, Category = "Event Tracing")
-	FString GDKEventPostFilter;
+	FString GDKEventPostFilter = "false";
 
 	/* The runtime filter which is used for local/cloud editor workflows (generated configs). */
 	UPROPERTY(EditAnywhere, Category = "Event Tracing")
-	FString RuntimeEventPreFilter;
+	FString RuntimeEventPreFilter = "false";
 
 	/* The runtime filter which is used for local/cloud editor workflows (generated configs). */
 	UPROPERTY(EditAnywhere, Category = "Event Tracing")
-	FString RuntimeEventPostFilter;
+	FString RuntimeEventPostFilter = "false";
+
+	FString GetGDKEventPreFilterString() const { return IsFilterValid(GDKEventPreFilter) ? GDKEventPreFilter : "false"; }
+	FString GetGDKEventPostFilterString() const { return IsFilterValid(GDKEventPostFilter) ? GDKEventPostFilter : "false"; }
+	FString GetRuntimeEventPreFilterString() const { return IsFilterValid(RuntimeEventPreFilter) ? RuntimeEventPreFilter : "false"; }
+	FString GetRuntimeEventPostFilterString() const { return IsFilterValid(RuntimeEventPostFilter) ? RuntimeEventPostFilter : "false"; }
+
+	TraceQueryPtr GetGDKEventPreFilter() const { return ParseOrDefault(GDKEventPreFilter, TEXT("gdk-pre-filter")); }
```

Comment:
Minor: you could consider an API based on enum instead of making one function per filter.